src/main/java/uk/gov/pay/directdebit/webhook/gocardless/services/handlers/GoCardlessEventHandler.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.pay.directdebit.webhook.gocardless.services.handlers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.directdebit.events.exception.GoCardlessEventHasNoMandateIdException;
+import uk.gov.pay.directdebit.events.exception.GoCardlessEventHasNoPaymentIdException;
+import uk.gov.pay.directdebit.events.model.GoCardlessEvent;
+import uk.gov.pay.directdebit.mandate.model.Mandate;
+import uk.gov.pay.directdebit.mandate.services.MandateQueryService;
+import uk.gov.pay.directdebit.notifications.services.UserNotificationService;
+import uk.gov.pay.directdebit.payments.exception.PaymentNotFoundException;
+import uk.gov.pay.directdebit.payments.model.GoCardlessPaymentId;
+import uk.gov.pay.directdebit.payments.model.Payment;
+import uk.gov.pay.directdebit.payments.services.PaymentQueryService;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.function.Predicate;
+
+import static uk.gov.pay.directdebit.events.model.GoCardlessResourceType.MANDATES;
+import static uk.gov.pay.directdebit.events.model.GoCardlessResourceType.PAYMENTS;
+import static uk.gov.pay.directdebit.events.model.GoCardlessResourceType.PAYOUTS;
+
+public class GoCardlessEventHandler {
```

Comment:
Since this class just deals with sending emails triggered by events now, could it have a more specific name?

Reply:
fixed. renamed to SendEmailsForGoCardlessEventsHandler